no-literal-checker/src/main/java/org/checkerframework/checker/noliteral/NoLiteralVisitor.java

```diff
@@ -0,0 +1,244 @@
+
+  @Override
+  protected void checkConstructorResult(
+      AnnotatedExecutableType constructorType, ExecutableElement constructorElement) {
+    // Skip this and do nothing. Normally, this method will issue a warning when
+    // the type of a class defaults to something other than top. That's fine
+    // for this checker, because we don't expect user-written classes to take
+    // on values other than @NonConstant.
+  }
+
+  /**
+   * The standard override checker is replaced so that no override.param.invalid errors are issued
+   * when overriding methods defined in bytecode (which therefore use a different, incompatible
+   * defaulting scheme).
+   */
+  @Override
+  protected OverrideChecker createOverrideChecker(
+      Tree overriderTree,
+      AnnotatedExecutableType overrider,
+      AnnotatedTypeMirror overridingType,
+      AnnotatedTypeMirror overridingReturnType,
+      AnnotatedExecutableType overridden,
+      AnnotatedTypeMirror.AnnotatedDeclaredType overriddenType,
+      AnnotatedTypeMirror overriddenReturnType) {
+    return new NoLiteralOverrideChecker(
+        overriderTree,
+        overrider,
+        overridingType,
+        overridingReturnType,
+        overridden,
+        overriddenType,
+        overriddenReturnType);
+  }
+
+  private class NoLiteralOverrideChecker extends OverrideChecker {
+    /**
+     * Create an OverrideChecker.
+     *
+     * <p>Notice that the return types are passed in separately. This is to support some types of
+     * method references where the overrider's return type is not the appropriate type to check.
+     *
+     * @param overriderTree the AST node of the overriding method or method reference
+     * @param overrider the type of the overriding method
+     * @param overridingType the type enclosing the overrider method, usually an
+     *     AnnotatedDeclaredType; for Method References may be something else
+     * @param overridingReturnType the return type of the overriding method
+     * @param overridden the type of the overridden method
+     * @param overriddenType the declared type enclosing the overridden method
+     * @param overriddenReturnType the return type of the overridden method
+     */
+    public NoLiteralOverrideChecker(
+        Tree overriderTree,
+        AnnotatedExecutableType overrider,
+        AnnotatedTypeMirror overridingType,
+        AnnotatedTypeMirror overridingReturnType,
+        AnnotatedExecutableType overridden,
+        AnnotatedTypeMirror.AnnotatedDeclaredType overriddenType,
+        AnnotatedTypeMirror overriddenReturnType) {
+      super(
+          overriderTree,
+          overrider,
+          overridingType,
+          overridingReturnType,
+          overridden,
+          overriddenType,
+          overriddenReturnType);
+    }
+
+    @Override
+    public boolean checkOverride() {
+      // Don't issue override errors if the overridden method was defined
+      // in bytecode, which uses a different defaulting scheme.
```

Comment:
If some method `a(int x)` is defined in a class `C` that is on the classpath but whose source is unavailable, and a class `D extends C` overrides `a(int x)`, then the version of `a` in `D` will be defaulted as if it is source code (that is, `x` will be `@NonConstant`, but the version of `a` in `C` will be defaulted as bytecode (i.e. `x` will be `@PolyConstant`. These types are incompatible, which is why the check is modified here.

I don't think this is a soundness concern. This code treats `C.a` as if its parameters are bottom and its return type is top - that is, treats it as if we were doing conservative defaulting. That means that the overriding method `D.a` has to follow the more restrictive conservative defaulting scheme, even though a call to `C.a` might be treated as polymorphic, which can only cause false positives (if `a` is called on an object whose static type is `C` but whose dynamic type is `D`).

At least, I think that's what's happening. It's been awhile since I wrote this code so I don't remember the original justification - this is my attempt to reconstruct it now.